build-tools/packages/build-cli/src/library/repoPolicyCheck/npmPackages.ts

```diff
@@ -1443,6 +1652,96 @@ export const handlers: Handler[] = [
 			return result;
 		},
 	},
+	{
+		// This rule enforces each exports type resolution (exported .d.ts file) is linted.
+		name: "npm-package-exports-apis-linted",
+		match,
+		handler: async (file: string): Promise<string | undefined> => {
+			let packageJson: PackageJson;
+
+			try {
+				packageJson = JSON.parse(readFile(file)) as PackageJson;
+			} catch {
+				return `Error parsing JSON file: ${file}`;
+			}
+
+			const { targetsImpacted } = await getApiLintElementsMissing(
+				packageJson,
+				path.dirname(file),
+			);
+
+			if (targetsImpacted.size > 0) {
+				return `exports types files ${[...targetsImpacted].join(
+					", ",
+				)} should be linted by check:exports:* script using api-extractor.`;
+			}
+		},
+		resolver: async (
```

Comment:
Asynchronous resolvers... how do you ensure that a resolver doesn't overwrite changes by other asynchronous resolvers?

Reply:
The tool scheduling the resolvers waits for each one. That already appeared to be the case.

Reply:
I think we might be in middle ground. routeToHandlers already has this code:
```typescript
						const { resolver } = handler;
						if (this.flags.fix && resolver) {
							output += `${newline}attempting to resolve: ${relPath}`;
							const resolveResult = await runWithPerf(handler.name, "resolve", async () =>
								resolver(relPath, gitRoot),
							);
```
that I didn't have to fix up.

But that doesn't really mean that everything is well scheduled. Looking at the surrounding code it says, if there is a handler that is async, then resolvers will run on arbitrary schedules.

Reply:
I split the async support to #21249